### PR TITLE
Add: [Win32] Select MIDI device by port name

### PR DESCRIPTION
**Pull request by nielsmh**
Also listing the available MIDI ports in debug on the win32_m driver, previously only the DMusic driver did that.

### CODE REVIEW

**Comment by glx22** [comment]
I get a different detection for dmusic and win32, but I guess it's expected.
```
dbg: [driver] Detected DirectMusic ports:
dbg: [driver]  0: Microsoft MIDI Mapper [Émulé]
dbg: [driver]  1: Microsoft GS Wavetable Synth [Émulé]
dbg: [driver]  2: Microsoft Synthesizer
dbg: [driver] Successfully probed music driver 'dmusic'

dbg: [driver] Win32-MIDI: Found 1 output devices:
dbg: [driver] MIDI port  0: Microsoft GS Wavetable Synth
dbg: [driver] Successfully loaded music driver 'win32'
```

**Comment by michicc** [comment]
Yeah, DMusic and normal Win synths are not identical or duplicates (which was the whole reason to make DMusic work on 64-bit as well).

**Comment by nielsmh** [comment]
The MIDI Mapper is not enumerated as a regular port in WinMM, you instead request it by a magical device ID. Actually, that may be one of the things that changed with the Mapper in one of the recent-ish Windows versions, maybe it was different in the old days.

File: src/music/win32_m.cpp [hunk 1]
```diff
@@ -370,6 +370,26 @@ const char *MusicDriver_Win32::Start(const char * const *parm)
...
 	int resolution = GetDriverParamInt(parm, "resolution", 5);
 	int port = GetDriverParamInt(parm, "port", -1);
+	const char *portname = GetDriverParam(parm, "portname");
+
+	/* Enumerate ports either for selecting port by name, or for debug output */
+	if (portname != nullptr || _debug_driver_level > 0) {
+		uint numports = midiOutGetNumDevs();
+		DEBUG(driver, 1, "Win32-MIDI: Found %d output devices:", numports);
+		for (uint tryport = 0; tryport < numports; tryport++) {
+			MIDIOUTCAPS moc{};
```
Comment (LordAro) [inline src/music/win32_m.cpp:380]: is there a particular need for {} ?

Comment (nielsmh) [reply src/music/win32_m.cpp:380]: It's probably not strictly required, mainly for being 110% sure nobody complains about passing pointers to uninitialised memory.

**Review by LordAro — APPROVED** [review]

**Pull request merged**